api/nodecontrol/client_test.go

```diff
@@ -0,0 +1,943 @@
+	"github.com/nats-io/nkeys"
+	"github.com/synadia-io/nex/api/nodecontrol/gen"
+	"github.com/synadia-io/nex/models"
+	"github.com/synadia-io/nex/node"
+)
+
+const (
+	// NCUU2YIYXEPGTCDXDKQR7LL5PXDHIDG7SDFLWKE3WY63ZGCZL2HKIAJT - pub key
+	Node1ServerSeed string = "SNAB2T3VG2363NDA2JK7NT5O3FN5VCXI2MYJHOPFO2NIDXQU6DIWQTBQC4"
+	// XAL54S5FE6SRPONXRNVE4ZDAOHOT44GFIY2ZW33DHLR2U3H2HJSXXRKY - pub xkey
+	Node1XKeySeed string = "SXAOUP7RZFW5QPE2GDWTPABUDM5UIAK6BPULJPWZQAFFL2RZ5K3UYWHYY4"
+)
+
+func buildTestBinary(t testing.TB, binMain string, workingDir string) (string, error) {
+	t.Helper()
+	binName := func() string {
+		if runtime.GOOS == "windows" {
+			return "test.exe"
+		}
+		return "test"
+	}
+
+	if _, err := os.Stat(filepath.Join(workingDir, binName())); err == nil {
+		return filepath.Join(workingDir, binName()), nil
+	}
+
+	cmd := exec.Command("go", "build", "-o", filepath.Join(workingDir, binName()), binMain)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	err := cmd.Run()
+	if err != nil {
+		return "", err
+	}
+
+	if _, err := os.Stat(filepath.Join(workingDir, binName())); err != nil {
+		return "", err
+	}
+
+	return filepath.Join(workingDir, binName()), nil
+}
+
+func startNatsServer(t testing.TB, workingDir string) (*server.Server, error) {
+	t.Helper()
+
+	s := server.New(&server.Options{
+		Port:      -1,
+		JetStream: true,
+		StoreDir:  workingDir,
+	})
+
+	s.Start()
+
+	go s.WaitForShutdown()
+
+	return s, nil
+}
+
+func startNexus(t testing.TB, ctx context.Context, logger *slog.Logger, workingDir, natsUrl string, numNodes int) error {
+	t.Helper()
+
+	nc, err := nats.Connect(natsUrl)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < numNodes; i++ {
+		var kp, xkp nkeys.KeyPair
+		if i == 0 {
+			kp, err = nkeys.FromSeed([]byte(Node1ServerSeed))
+			if err != nil {
+				return err
+			}
+			xkp, err = nkeys.FromSeed([]byte(Node1XKeySeed))
+			if err != nil {
+				return err
+			}
+		} else {
+			kp, err = nkeys.CreateServer()
+			if err != nil {
+				return err
+			}
+			xkp, err = nkeys.CreateCurveKeys()
+			if err != nil {
+				return err
+			}
+		}
+		nn, err := node.NewNexNode(kp, nc,
+			models.WithContext(ctx),
+			models.WithLogger(logger),
+			models.WithXKeyKeyPair(xkp),
+			models.WithNodeName(fmt.Sprintf("node-%d", i+1)),
+			models.WithNexus("testnexus"),
+			models.WithResourceDirectory(workingDir),
+		)
+		if err != nil {
+			return err
+		}
+
+		err = nn.Validate()
+		if err != nil {
+			return err
+		}
+
+		go func() {
+			err = nn.Start()
+			if err != nil {
+				t.Error(err)
+				t.FailNow()
+			}
+		}()
+	}
+
+	return nil
+}
+
+func TestAuction(t *testing.T) {
+	workingDir := t.TempDir()
+	natsServer, err := startNatsServer(t, workingDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+
+	t.Cleanup(func() {
+		os.RemoveAll(filepath.Join(os.TempDir(), "inex-NCUU2YIYXEPGTCDXDKQR7LL5PXDHIDG7SDFLWKE3WY63ZGCZL2HKIAJT"))
```

Comment:
Since the working directory is created from `t.TempDir()`, it will be cleaned up automatically when the test ends. No need for a separate cleanup call.

Reply:
the `inex-*` directory is actually in os.TempDir and not configurable at this point...it is designed to persist incase of a node failure

Reply:
To confirm this doesn't correspond to the `WithResourceDirectory` option for the node?

Reply:
It does not....it probably should

Reply:
https://github.com/synadia-io/nex/issues/474